mongorestore/mongorestore_test.go

```diff
@@ -487,3 +487,110 @@ func TestKnownCollections(t *testing.T) {
 		})
 	})
 }
+
+
+func TestFixHashedIndexes(t *testing.T) {
```

Comment:
There were more test cases in the mongomirror PR that it would be nice to add: https://github.com/10gen/mongomirror/pull/186/files#diff-15b0cb4c08056e459d5b9cb4388e1403R6542

Reply:
Those test cases seem pretty unnecessary to me, but I don't mind adding them.

Reply:
Well, the one testing that a non-dotted hashed index is maintained _is_ good, though!

Reply:
So, adding the one for the non-dotted hashed index. I don't think there is much reason to see that normal indexes are not randomly changed, nor text.

Reply:
Sounds fair to me as long as the others agree :)

Reply:
Seems fine to me!

Reply:
The line on 359 has `A && B` logic, so there are three cases to test: A false; A true and B false, and both A and B true.  I'd like to have all three covered, but each only needs to be covered once.  :-)

Reply:
I think you are right, I'll do that. Easy enough to add an index for `a.a` that is 1 already.